### PR TITLE
Endpoint -> Environment saved to a yaml doc

### PR DESCRIPTION
**Pull request by Squab**
Allows for saving environment data in a yaml file.
This should be easily extendable for setting general environment variables,
not just headers and endpoint

### CODE REVIEW

**Comment by Squab** [comment]
Damn you python3, will fix tomorrow

File: spag.py [hunk 1]
```diff
@@ -7,37 +7,49 @@
...
+import yaml
 
 class ToughNoodles(Exception):
     pass
 
-class Endpoint(object):
+class Environment(object):
 
-    ENDPOINT_FILE = '.spag.endpoint'
+    ENV_FILE = '.env.yml'
```
Comment (pglass) [inline spag.py]: We should make this `.spag.env` or something spag-specific.

File: spag.py [hunk 2]
```diff
@@ -7,37 +7,49 @@
...
+
+        if os.path.exists(cls.ENV_FILE):
+            f = click.open_file(cls.ENV_FILE, 'r')
+            data = yaml.load(f)
+            data.update(kwargs)
+        else:
+            data = kwargs
+
+        f = click.open_file(cls.ENV_FILE, 'w+')
+        yaml.safe_dump(data, f)
```
Comment (pglass) [inline spag.py]: This looks like creates a `.env.yaml` file that contains `{endpoint: poo}` instead of just `endpoint: poo`?

Comment (pglass) [inline spag.py]: Apparently, if you `yaml.dump(x, default_flow_style=False)` it will get rid of the braces.

**Pull request merged**